evoting/lib/election.go

```diff
@@ -75,30 +77,28 @@ func GetElection(roster *onet.Roster, id skipchain.SkipBlockID) (*Election, erro
 		return nil, fmt.Errorf("no election structure in %s", id.Short())
 	}
 	election := transaction.Election
-	err = election.setStage()
+	err = election.setStage(s)
 	if err != nil {
 		return nil, err
 	}
 	return election, nil
 }
 
-func (e *Election) setStage() error {
-	mixes, err := e.Mixes()
-	if err != nil {
-		return err
-	}
-
-	partials, err := e.Partials()
+func (e *Election) setStage(s *skipchain.Service) error {
+	db := s.GetDB()
+	latest, err := db.GetLatest(db.GetByID(e.ID))
 	if err != nil {
-		return err
+		return errors.New("error getting latest skipblock")
 	}
+	transaction := UnmarshalTransaction(latest.Data)
 
-	if len(mixes) == 0 {
-		e.Stage = Running
-	} else if len(partials) == len(e.Roster.List) {
+	// TODO: check for length and check if chain is corrupt
```

Comment:
We can suppose that the skipchain service is not corrupt. Before the skipchain service stores a new block in its database, it checks that it is a valid block.